client/boot/index.js

```diff
@@ -361,6 +362,20 @@ function reduxStoreReady( reduxStore ) {
 		require( 'lib/rubberband-scroll-disable' )( document.body );
 	}
 
+	/*
+	 * Layouts with differing React mount-points will not reconcile correctly,
+	 * so remove an existing single-tree layout by re-rendering if necessary.
+	 *
+	 * TODO (@seear): React 15's new reconciliation algo may make this unnecessary
+	 */
+	page( '*', function( context, next ) {
+		if ( ! isIsomorphicRoute( context.path ) && previousLayoutIsSingleTree() ) {
```

Comment:
> we would get this re-render on any transition from layout-last route to layout-last route, e.g. logged-out /design to /theme.

Right, that makes sense. Unlike my previous comment where I was ignoring the `!` :-D Probably a good idea to keep it then.

Cleaner to use a selector though? It's in 3633d06a6d8acda6f461415596a3a2b296a4222a (be sure to pull first), should be possible to just `cherry-pick` that commit then...